docs/src/main/sphinx/sql/merge.rst

```diff
@@ -0,0 +1,94 @@
+=====
+MERGE
+=====
+
+Synopsis
+--------
+
+.. code-block:: text
+
+    MERGE INTO target_table [ [ AS ]  target_alias ]
+    USING { source_table | query } [ [ AS ] source_alias ]
+    ON search_condition
+    when_clause [...]
+
+where ``when_clause`` is one of
+
+.. code-block:: text
+
+    WHEN MATCHED [ AND condition ]
+        THEN DELETE
+
+.. code-block:: text
+
+    WHEN MATCHED [ AND condition ]
+        THEN UPDATE SET ( column = expression [, ...] )
+
+.. code-block:: text
+
+    WHEN NOT MATCHED [ AND condition ]
+        THEN INSERT [ column_list ] VALUES (expression, ...)
+
+Description
+-----------
+
+Conditionally update and/or delete rows of a table and/or insert new
+rows into a table.
+
+``MERGE`` supports an arbitrary number of ``WHEN`` clauses with different
+``MATCHED`` conditions, executing the ``DELETE``, ``UPDATE`` or ``INSERT``
+operation in the first ``WHEN`` clause selected by the ``MATCHED``
+state and the match condition.
+
+For each source row, the ``WHEN`` clauses are processed in order.  Only
+the first first matching ``WHEN`` clause is executed and subsequent clauses
+are ignored.  A ``MERGE_TARGET_ROW_MULTIPLE_MATCHES`` exception is
```

Comment:
We should mention what happens if not clauses match the row and there is no default `WHEN MATCHED` clause

Reply:
Added:

```
If a source row is not matched by any ``WHEN`` clause and there is no
``WHEN NOT MATCHED` clause, the source row is ignored.
```